module/VuFind/src/VuFind/Sitemap/Generator.php

```diff
@@ -347,62 +275,55 @@ protected function generateWithPlugins(): array
             &$sitemapFiles,
             &$sitemapIndexes
         ) {
-            $index = $sitemapIndexes[$name] ?? 0;
-            ++$index;
+            $index = ($sitemapIndexes[$name] ?? 0) + 1;
             $sitemapIndexes[$name] = $index;
-            $pageName = "$name-$index";
+            $pageName = empty($name) ? $index : "$name-$index";
             $filePath = $this->getFilenameForPage($pageName);
             if (false === $sitemap->write($filePath)) {
                 throw new \Exception("Problem writing $filePath.");
             }
             $sitemapFiles[] = $this->getFilenameForPage($pageName, false);
         };
 
-        if ($plugins = $this->config->Sitemap->plugins ?? []) {
-            $pluginSitemaps = [];
-            foreach ($plugins->toArray() as $pluginName) {
-                $plugin = $this->getPlugin($pluginName);
-                $sitemapName = $plugin->getSitemapName();
-                $this->verboseMsg(
-                    "Generating sitemap '$sitemapName' with '$pluginName'"
-                );
-                if (!isset($pluginSitemaps[$sitemapName])) {
-                    $pluginSitemaps[$sitemapName] = $this->getNewSitemap();
-                }
-                $languages = $plugin->supportsVuFindLanguages()
-                    ? $this->languages : [];
-                $frequency = $plugin->getFrequency();
-                $sitemap = &$pluginSitemaps[$sitemapName];
-                $count = $sitemap->getCount();
-                foreach ($plugin->getUrls() as $url) {
-                    ++$count;
-                    if ($count > $this->countPerPage) {
-                        // Write the current sitemap and clear all entries from it:
-                        $writeMap($sitemap, $sitemapName);
-                        $sitemap->clear();
-                        $count = 1;
-                    }
-                    if (($languages || $frequency) && is_string($url)) {
-                        $sitemap->addUrl(
-                            [
-                                'url' => $url,
-                                'languages' => $languages,
-                                'frequency' => $frequency,
-                            ]
-                        );
-                    } else {
-                        $sitemap->addUrl($url);
-                    }
-                }
-                // Unset the reference:
-                unset($sitemap);
+        $mandatoryPlugins = ['Index'];
```

Comment:
I was worried that this would cause backward-incompatibility headaches with #1904, but since #1904 actually hasn't been released yet (it's new to 8.0), this actually is no problem at all -- a warning note to vufind-tech should more than cover it. I'll make the change as soon as time permits!

Reply:
Okay, I took care of this in 70975c3 -- if you have no further suggestions, I'll see if I can find time to write a couple of unit tests, and then will merge after that.